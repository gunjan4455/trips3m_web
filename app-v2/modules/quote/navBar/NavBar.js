import React from 'react';
import PropTypes from 'prop-types';
import Sticky from 'react-stickynode';

import DetailNav from 'modules/shared/DetailNav';

const NavBar = ({ options, trackSegment }) => {
  return (
    <div className="mt8">
      <Sticky enabled={true} top={0} innerZ={4}>
        <DetailNav options={options} trackSegment={trackSegment} />
      </Sticky>
    </div>
  );
};

NavBar.propTypes = {
  options: PropTypes.array.isRequired,
  trackSegment: PropTypes.func
};

NavBar.defaultProps = {
  trackSegment: () => {}
};

export default NavBar;
