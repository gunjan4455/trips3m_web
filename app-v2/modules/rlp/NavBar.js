import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Notification from 'modules/notifications';
import BurgerMenu from 'modules/header/BurgerMenu';

class NavBar extends Component {
  static propTypes = {
    currentTripId: PropTypes.number.isRequired,
    trackSegment: PropTypes.func.isRequired
  };

  render() {
    const NotificationContainer = Notification.container;
    return (
      <div className="pbc5 p8">
        <div className="pt4 pb4 pr8 pl8 flex wfull justifyCenter alignCenter">
          <BurgerMenu />
          <p className="pl15 f16 flex alignCenter fw7 sfcw flexFull ellipsis">
            My Trip
          </p>
          <NotificationContainer />
        </div>
      </div>
    );
  }
}

export default NavBar;
