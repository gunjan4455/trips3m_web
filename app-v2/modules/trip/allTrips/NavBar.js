import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Back2IconWhite } from 'helpers/Icon/Icon';

class NavBar extends Component {
  static propTypes = {
    goBack: PropTypes.func,
    trackSegment: PropTypes.func
  };

  static defaultProps = {
    goBack: () => {},
    trackSegment: () => {}
  };

  render() {
    return (
      <div className="pbc5 p8">
        <div className="pt4 pb4 pr8 pl8 flex wfull justifyCenter alignCenter">
          <button
            onClick={this.props.goBack}
            className="input-24 block mr15 at_goback"
          >
            <Back2IconWhite />
          </button>
          <p className="f16 flex alignCenter fw7 sfcw flexFull ellipsis">
            All Trips
          </p>
        </div>
      </div>
    );
  }
}

export default NavBar;
