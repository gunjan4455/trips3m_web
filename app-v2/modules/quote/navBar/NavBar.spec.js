import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import NavBar from './NavBar';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  // TODO: Rahul- Add options here
  options: [],
};

const shallowWrappperNavBar = (props) => {
  return shallow(<NavBar {...props} />);
};

describe('NavBar: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperNavBar(props);
    expect(wrapper).toMatchSnapshot();
  });

});
