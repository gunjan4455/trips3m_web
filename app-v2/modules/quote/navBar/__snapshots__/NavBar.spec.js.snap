// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    options={Array []}
    trackSegment={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Sticky
        activeClass="active"
        bottomBoundary={0}
        enableTransforms={true}
        enabled={true}
        innerZ={4}
        onStateChange={null}
        releasedClass="released"
        shouldFreeze={[Function]}
        top={0}
      >
        <DetailNav
          options={Array []}
          trackSegment={[Function]}
        />
      </Sticky>,
      "className": "mt8",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeClass": "active",
        "bottomBoundary": 0,
        "children": <DetailNav
          options={Array []}
          trackSegment={[Function]}
        />,
        "enableTransforms": true,
        "enabled": true,
        "innerZ": 4,
        "onStateChange": null,
        "releasedClass": "released",
        "shouldFreeze": [Function],
        "top": 0,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "options": Array [],
          "trackSegment": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Sticky
          activeClass="active"
          bottomBoundary={0}
          enableTransforms={true}
          enabled={true}
          innerZ={4}
          onStateChange={null}
          releasedClass="released"
          shouldFreeze={[Function]}
          top={0}
        >
          <DetailNav
            options={Array []}
            trackSegment={[Function]}
          />
        </Sticky>,
        "className": "mt8",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeClass": "active",
          "bottomBoundary": 0,
          "children": <DetailNav
            options={Array []}
            trackSegment={[Function]}
          />,
          "enableTransforms": true,
          "enabled": true,
          "innerZ": 4,
          "onStateChange": null,
          "releasedClass": "released",
          "shouldFreeze": [Function],
          "top": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "options": Array [],
            "trackSegment": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
